endpoint-sec/src/client.rs

```diff
@@ -667,10 +669,13 @@ impl Client {
 }
 
 /// Private helper methods
-impl Client {
+impl Client<'_> {
     /// Construct a client from a raw pointer.
-    fn from_raw(inner: NonNull<es_client_t>) -> Self {
-        Self { inner }
+    fn from_raw<'b>(inner: NonNull<es_client_t>) -> Client<'b> {
```

Comment:
hol' up how is that not `unsafe` ?

Also, this function is only used in one spot afaict, in `fn new`. Could we just... pull its content there? It will feel a bit less spooky there.